ls1intum/Artemis
# `Development`: Remove unnecessary path parameters

### Checklist
#### General


#### Server
#### Client

### Motivation and Context
The FileResource class unnecessarily uses some additional path parameters, which reduces simplicity and increases the chances of bugs and/or missuse.

### Description
Removed the aforementioned parameters from the paths.

### Steps for Testing
We just want to make sure that I didn't accidentally break the implementation.

Prerequisites:
- 1 Instructor
- 1 Student

Course (Instructor/Student):
1. Look at a course with course image (e.g. 6.0 Testing Session Course 1 on TS1). Alternatively, create one
2. Make sure the image gets rendered correctly

After each step, commit and see if the changes were persisted by reloading
Programming Exercise (Instructor, Student):
1. Create a programming exercise with the online editor activated.
2. Change the programming language and make sure the template problem statement gets changed.
3. Actually create the exercise (submit) and make sure all repos got created correctly (tests should return correctly for solution and template.
4. Feel free to check a simple student submission too
5. No issues should occur

Fileupload Exercise (Instructor, Student):
1. Create a file upload exercise with a description. Upload an image to the description
2. Start the exercise, the image should show up properly as well as the rest of the description
3. Participate in the exercise
4. Upload another file
5. View/Download your submission
6. No issues should occur, the correct file should download.

Quiz Exercise (Instructor):
1. Create a Quiz Exercise with a Drag and Drop Question including Background and image drag items
2. Preview the quiz and make sure all images get rendered properly

#### Exam Mode Testing



We would need to test the participation checker which requires testing it on an app. However, the changes to the two endpoints are the exact same changes I did to a few other endpoints. Meaning if one fails, all should fail.
=> I deem it unnecessary to test every single of these endpoints. They already get tested by the other endpoints.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






I still added the coverage here, but there were no functional changes only some small clean-up which is covered, so I did not add more tests for things I didn't touch in this PR.
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| FileResource.java | 70.2% |                            |
| programming-exercise-update.component.ts | 79.9% |               |
| code-editor-file-browser.component.ts | 85.4% |               |
| file.service.ts | 47.3% |               |
